### DISCUSSION TITLE
AutoDarkModeSvc startup impact high

### DISCUSSION THREAD

**Discussion by lambdakilo** [AutoDarkMode/Windows-Auto-Night-Mode#753] [Q&A]
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/31825085/1d21bb02-83d1-4731-970c-12542a69adba)
Does anyone else have their startup impact marked as high on their Windows? Could there be something done about it inside code?

**Comment by Armin2208** [comment]
That's normal, because Auto Dark Mode runs with a full .Net-Runtime at startup.

**Reply by lambdakilo** [reply]
Thanks for the clarification. How time-consuming approximately it would be to rewrite ADM with a lighter environment?

**Reply by Armin2208** [reply]
Auto Dark Mode was already once rewritten. We decided to switch from .Net Framework to .Net Core. Because it's Microsofts new thing with all the new platform features. The real fix would be Microsoft including the .Net Core Runtime into Windows. So we wouldn't need to ship the whole Runtime with ADM. This would also help many other Windows developers. We will not switch to any other environment.
